crates/ruma-events/src/invite_permission_config.rs

```diff
@@ -0,0 +1,317 @@
+    }
+
+    /// Whether all invites are allowed according to this account data.
+    fn are_all_invites_allowed(&self) -> bool {
+        self.ignored_users.is_empty()
+            && self.blocked_users.is_empty()
+            && self.ignored_servers.is_empty()
+            && self.blocked_servers.is_empty()
+    }
+
+    /// Get the decision for the given invite sender according to this account data.
+    pub fn decision_for_sender(&self, sender: &UserId) -> InvitePermissionDecision {
+        if self.are_all_invites_allowed() {
+            return InvitePermissionDecision::Allow;
+        }
+
+        fn list_has_match(list: &[String], string: &str) -> bool {
+            list.iter().any(|pattern| WildMatch::new(pattern).matches(string))
+        }
+
+        let sender_str = sender.as_str();
+
+        if list_has_match(&self.allowed_users, sender_str) {
+            return InvitePermissionDecision::Allow;
+        }
+
+        if list_has_match(&self.ignored_users, sender_str) {
+            return InvitePermissionDecision::Ignore;
+        }
+
+        if list_has_match(&self.blocked_users, sender_str) {
+            return InvitePermissionDecision::Block;
+        }
+
+        let server_name = sender.server_name().as_str();
+
+        if list_has_match(&self.allowed_servers, server_name) {
+            return InvitePermissionDecision::Allow;
+        }
+
+        if list_has_match(&self.ignored_servers, server_name) {
+            return InvitePermissionDecision::Ignore;
+        }
+
+        if list_has_match(&self.blocked_servers, server_name) {
+            return InvitePermissionDecision::Block;
+        }
+
+        InvitePermissionDecision::Allow
+    }
+}
+
+/// The unstable version of [`InvitePermissionConfigEventContent`], using the unstable prefix
+/// defined in [MSC4155].
+///
+/// [MSC4155]: https://github.com/matrix-org/matrix-spec-proposals/pull/4155
+#[derive(Clone, Debug, Default, Deserialize, Serialize, EventContent)]
+#[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+#[ruma_event(type = "org.matrix.msc4155.invite_permission_config", kind = GlobalAccountData)]
+#[serde(transparent)]
+pub struct UnstableInvitePermissionConfigEventContent(pub InvitePermissionConfigEventContent);
+
+impl Deref for UnstableInvitePermissionConfigEventContent {
+    type Target = InvitePermissionConfigEventContent;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl From<UnstableInvitePermissionConfigEventContent> for InvitePermissionConfigEventContent {
+    fn from(value: UnstableInvitePermissionConfigEventContent) -> Self {
+        value.0
+    }
+}
+
+impl From<InvitePermissionConfigEventContent> for UnstableInvitePermissionConfigEventContent {
+    fn from(value: InvitePermissionConfigEventContent) -> Self {
+        Self(value)
+    }
+}
+
+/// The decision for an invite according to an [`InvitePermissionConfigEventContent`].
+#[derive(Debug, Default, Clone, Copy, PartialEq, Eq)]
```

Comment:
Why have a `Default`?

Reply:
> Maybe users should use instead the `Default` implementation of account data if it is missing?

Probably? But anyways, your reasoning also makes sense for having a `Default`, so we can keep it if you want.